### PR TITLE
Add shape_env guards to tracing context

### PR DESCRIPTION
**Pull request by voznesenskym**
cc @mlazos @soumith @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -765,15 +765,15 @@ def evaluate_guards_for_args(self, placeholders, args):
         return eval(code, {}, dict(zip(arg_names, args)))
 
     def get_nontrivial_guards(self):
-        return [self.simplify(guard) for guard, _ in self.guards if self._maybe_evaluate_static(guard) is None]
+        return [self.simplify(guard.expr) for guard in self.guards if self._maybe_evaluate_static(guard.expr) is None]
```
Comment (Chillee) [inline torch/fx/experimental/symbolic_shapes.py:768]: Why are these changing?

**Review by Chillee — APPROVED** [review]
Some extra breakpoints left over

File: torch/_dynamo/symbolic_convert.py [hunk 2]
```diff
@@ -319,6 +319,7 @@ def wrapper(self: "InstructionTranslatorBase", inst: Instruction):
                 user_stack = [self.frame_summary()] + list(reversed(excp.real_stack))
                 user_stack_formatted = "".join(traceback.format_list(user_stack))
                 frame_loc = (user_stack[-1].filename, user_stack[-1].lineno)
+                breakpoint()
```
Comment (voznesenskym) [inline torch/_dynamo/symbolic_convert.py:322]: lol woops

File: torch/fx/experimental/symbolic_shapes.py (hunk 1 above)
Comment (voznesenskym) [reply torch/fx/experimental/symbolic_shapes.py:768]: Readability

Comment (voznesenskym) [reply torch/fx/experimental/symbolic_shapes.py:768]: to elaborate: I could rename this from guard to something like:

`return [self.simplify(expr) for expr, _ in self.guards if self._maybe_evaluate_static(expr) is None]`

But that felt less readable than this

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**